### ISSUE TITLE
pipelines: Find a better solution for deleting the `_the..` after Windows builds

### ISSUE THREAD

**Issue by Willsparker** [AdoptOpenJDK/openjdk-build#2205]
ref: #2204 , #1956 , https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1662

With the 2 linked PRs above, we have put in manual ways of deleting the `_the..` file, however, the way it's being done is quite hacky. We should probably see why / where the locked file comes from or at least put in a more universal way of deleting the file.

Some additional: https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1396

**Comment by Haroon-Khel** [comment]
Wouldnt the following suffice?

```
if (buildConfig.TARGET_OS == "windows") {
                  context.sh(script: "rm -rf $CYGWIN_WORKSPACE/workspace/build/src/build/*/jdk/gensrc")
}
```
And perhaps a check to see if $CYGWIN_WORKSPACE exists, or is nonempty?

**Comment by Willsparker** [comment]
I tried something similar when this first cropped up : https://github.com/AdoptOpenJDK/openjdk-build/pull/1956#issuecomment-654163802 
I would have put in a general fix in for #1956 , but unfortunately I don't know the variables that are available in the groovy script, at that part of the build, and anything I tried didn't seem to work. 
@M-Davies is there a groovy version of `printenv` that we could put in as debug to see what variables there are?

**Comment by M-Davies** [comment]
[There is an `env` variable that is passed in from `kick_off_build.groovy` you could print out?](https://github.com/AdoptOpenJDK/openjdk-build/blob/master/pipelines/build/common/openjdk_build_pipeline.groovy#L83). We also have some other variables declared on the server itself https://ci.adoptopenjdk.net/pipeline-syntax/globals

**Comment by Willsparker** [comment]
AH alright. I'll try putting some debug stuff to print the contents of `env`, and the global pipeline variable `WORKSPACE`.

**Comment by Willsparker** [comment]
Fixed as of #2416
